src/utils.js

```diff
@@ -35,15 +38,273 @@ playFeedbackAudio(true, 'correct.mp3', 'incorrect.mp3')
 playFeedbackAudio(null, 'neutral.mp3',)
 */
 
-
 export function playFeedbackAudio(responseIsCorrect, audio1, audio2) {
-    let audioToPlay
+  let audioToPlay
+
+  if (responseIsCorrect || responseIsCorrect === null) {
+    audioToPlay = audio1
+  } else {
+    audioToPlay = audio2
+  }
+
+  new Audio(audioToPlay).play()
+}
+
+// Gets the app language based on the lng query string. Using this because ROAR apps use i18next.
+function getLanguage(setLanguage) {
+  if (setLanguage) return setLanguage
+
+  let results = new RegExp('[\?&]lng=([^&#]*)').exec(window.location.href);
+  if (results == null) {
+    // default to English
+     return 'en';
+  }
+  return decodeURI(results[1]) || 0;
+}
+
+// Gets the device type based on the device inputs
+function getDevice() {
+  if (deviceType === 'touchOnly' || (deviceType === 'hybrid' && primaryInput === 'touch')) {
+    return 'mobile'
+  }
+  return 'desktop'
+}
+
+function getFormattedURL(bucketURI, filePath, lng, device, type, nested, isDefault) {
+  if ((isDefault && nested && type === 'device') || (nested && type === 'device')) {
+      return `${bucketURI}/${lng}/${device}/${filePath}`;
+  } else if ((isDefault && nested && type === 'shared') || (nested && type === 'shared')) {
+      return `${bucketURI}/${lng}/shared/${filePath}`;
+  } else if (type === 'device') {
+      return `${bucketURI}/${device}/${filePath}`;
+  } else if (type === 'shared/device') {
+      return `${bucketURI}/shared/${device}/${filePath}`;
+  } else if (type === 'languageSpecific') {
+      return `${bucketURI}/${lng}/${filePath}`;
+  } else if (type === 'default') {
+      return `${bucketURI}/${filePath}`;
+  } else {
+      return `${bucketURI}/shared/${filePath}`;
+  }
+}
+
+function getAssetType(asset) {
+  const mimeType = mime.lookup(asset);
+  if (!mimeType) {
+      throw new Error(`Unrecognized file extension in path: ${asset}`);
+  }
+  if (mimeType.startsWith('image/')) return 'images';
+  if (mimeType.startsWith('audio/')) return 'audio';
+  if (mimeType.startsWith('video/')) return 'video';
+  throw new Error(`Unsupported MIME type for file: ${asset}. Only image, audio, and video files are supported.`);
+}
+
+
+
+// Takes in a JSON file, a cloud storage bucket URI, and optionally a language code and returns an object with the the asset paths mapped to thier file names, categorized by media type.
+// Ex.
+// {
+//   images: {
+//     image1: 'path/to/image1.jpg'
+//   }
+//   audio: {
+//     audio1: 'path/to/audio1.mp3'
+//   }
+// }
+
```

Comment:
This could all go in the jsDoc string. The docstring here probably needs to be fairly extensive to describe both the intended JSON input and the expected output.